gemrb/core/Scriptable/Scriptable.cpp

```diff
@@ -204,7 +208,11 @@ void Scriptable::SetScript(int index, GameScript* script)
 		Log(ERROR, "Scriptable", "Invalid script index!");
 		return;
 	}
-	delete Scripts[index];
+	if (Scripts[index] && Scripts[index]->running) {
+		Scripts[index]->dead = true;
+	} else {
+		delete Scripts[index];
+	}
```

Comment:
I've removed this function yesterday, since it's unused, just haven't pushed, but I think git will be smart enough to not care.